src/main/java/com/mongodb/jdbc/MongoDriver.java

```diff
@@ -153,39 +151,59 @@ protected MongoConnection getUnvalidatedConnection(String url, Properties info)
             info = new Properties();
         }
 
-        // reuse the code getPropertyInfo to make sure the URI is properly set wrt the passed
+        // Ensure that the ConnectionString and Properties are consistent.
+        // Reuse the code getPropertyInfo to make sure the URI is properly set wrt the passed
         // Properties info value.
-        Pair<ConnectionString, DriverPropertyInfo[]> p = getConnectionString(url, info);
-        // ensure that the ConnectionString and Properties are consistent.
-        reconcileProperties(p.right(), info);
-
-        ConnectionString cs = p.left();
-        return createConnection(cs, info);
+        Pair<ConnectionString, DriverPropertyInfo[]> p = getConnectionSettings(url, info);
+        checkMissingProperties(p.right());
+        return createConnection(p.left(), info);
     }
 
-    private void reconcileProperties(DriverPropertyInfo[] driverPropertyInfo, Properties info)
+    /**
+     * Provides feedback regarding the missing properties.
+     *
+     * @param missingRequiredProperties
+     * @throws SQLException The connection error to return to the user.
+     */
+    private void checkMissingProperties(DriverPropertyInfo[] missingRequiredProperties)
             throws SQLException {
         // since the user is calling connect, we should throw an SQLException if we get
         // a prompt back. Inspect the return value to format the SQLException.
-        if (driverPropertyInfo.length != 0) {
-            if (driverPropertyInfo[0].name.equals(USER)) {
-                throw new SQLException("password specified without user");
-            }
-            if (driverPropertyInfo[0].name.equals(PASSWORD)) {
-                throw new SQLException("user specified without password");
-            }
-            String[] propertyNames = new String[driverPropertyInfo.length];
-            for (int i = 0; i < propertyNames.length; ++i) {
-                propertyNames[i] = driverPropertyInfo[i].name;
+        if (missingRequiredProperties.length != 0) {
+            for (DriverPropertyInfo info : missingRequiredProperties) {
```

Comment:
Not sure I understand the logic here.  There is a for loop through the properties, however it looks like the first one will always throw the  SQLException.  Or was it meant to loop through the properties and append the errors to a string and throw that exception at the end?